pkg/kubelet/images/image_gc_manager_test.go

```diff
@@ -548,14 +550,19 @@ func TestValidateImageGCPolicy(t *testing.T) {
 	}
 }
 
+//
 func TestImageCacheReturnCopiedList(t *testing.T) {
 	cache := &imageCache{}
 	testList := []container.Image{{ID: "1"}, {ID: "2"}}
+	// Make a copy of testList because cache.set takes ownership of the list
+	testListCopy := append([]container.Image{}, testList...)
+	// Expect cache.get to sort the image list
+	sort.Sort(sliceutils.ByImageSize(testListCopy))
 	cache.set(testList)
```

Comment:
We should also test that `testList` is not changed after `get`.

Reply:
It is changed. That's why the test didn't catch the bug before. Please see the comment I added for `imageCache.set()`

It brings another question: should `imageCache.set()` make a copy of the given list? E.g.
```
func (i *imageCache) set(images []container.Image) {
	i.Lock()
	defer i.Unlock()
	imagesCopy := append([]container.Image{}, images...)
	i.images = imagesCopy
}
```

this way the caller of `imageCache.set()` can safely reference `images` afterwards. I think it's a reasonable change but not as critical. Currently [this goroutine](https://github.com/kubernetes/kubernetes/blob/1836f952600c08ac3bdc21e38c72bd5d28135ed5/pkg/kubelet/images/image_gc_manager.go#L194) is the only caller of `imageCache.set()` and it doesn't reference `images` afterwards, so the change doesn't matter. Since we are going to cherrypick this fix, I prefer keeping the minimum change needed. That's why I didn't change `imageCache.set()` behavior here.

Reply:
I moved the sort logic outside of the image cache and tested that `testList` is not changed after `get`